grails-app/views/inventoryLevel/_form.gsp

```diff
@@ -32,10 +32,11 @@
                     <label for="inventory.id"><warehouse:message code="inventoryLevel.inventory.label" default="Inventory" /></label>
                 </td>
                 <td valign="top" class="value ${hasErrors(bean: inventoryLevelInstance, field: 'inventory', 'errors')}">
-                    <g:select id="${inventoryLevelInstance?.id?'edit':'save'}-${inventoryLevelInstance?.id}-inventory-id" name="inventory.id" from="${org.pih.warehouse.inventory.Inventory.list()}" optionKey="id"
+                    <g:set var="location" value="${Inventory.list().find { it.warehouse == currentLocation } }"/>
```

Comment:
This is bad for several reasons. 
1. You're doing database queries in the view, which is bad practice. At the very least you should do this in the controller and pass the `inventory` in the model. 
2. This is very inefficient. The list() method is a `select * from table` query so it can be expensive. I've already refactored a few places where you've done this, so please don't perform lookups like this anywhere in the code. If you find yourself needing to use list() to iterate through results to get a single object, please ask me to write a query for you. 

Your options are:
1. Lookup inventory by location
    ```
    def inventory = Inventory.findByLocation(currentLocation) 
    ```
1. Get the inventory directly from the location (recommended)
    ```
    def location = Location.get(session.warehouse.id)
    def inventory = location?.inventory
    ```